src/marqo/tensor_search/tensor_search.py

```diff
@@ -1539,27 +1579,76 @@ def get_query_vectors_from_jobs(queries: List[BulkSearchQueryEntity], qidx_to_jo
         - config: standard Marqo config.
 
     """
-    result: Dict[Qidx, List[List[float]]] = defaultdict(list)
-    for qidx, ptr in qidx_to_job.items():
-        vectors = job_to_vectors[ptr.job_hash][ptr.start_idx: ptr.end_idx]
-        # qidx_to_vectors[qidx].append(vectors)
+    result: Dict[Qidx, List[float]] = defaultdict(list)
+    for qidx, ptrs in qidx_to_job.items():
+
+        # vectors = job_to_vectors[ptrs.job_hash][ptrs.start_idx: ptrs.end_idx]
 
+        # qidx_to_vectors[qidx].append(vectors)
         q = queries[qidx]
+        index_info = get_index_info(config=config, index_name=q.index)
+
         ordered_queries = list(q.q.items()) if isinstance(q.q, dict) else None
         if ordered_queries:
             # multiple queries. We have to weight and combine them:
-            index_info = get_index_info(config=config, index_name=q.index)
-            weighted_vectors = [np.asarray(vec) * weight for vec, weight in zip(vectors, [w for _, w in ordered_queries])]
+            vectorised_ordered_queries = [
+                (get_content_vector(
+                    possible_jobs=qidx_to_job[qidx],
+                    jobs=jobs,
+                    job_to_vectors=job_to_vectors,
+                    treat_urls_as_images=index_info.index_settings[NsField.index_defaults][NsField.treat_urls_and_pointers_as_images],
+                    content=content),
+                 weight,
+                 content
+                ) for content, weight in ordered_queries
+            ]
+            # TODO how doe we ensure order?
```

Comment:
Yeah, I think a big question of going from `qidx_to_job: Dict[Qidx, VectorisedJobPointer]` to `Dict[Qidx, List[VectorisedJobPointer]]` is how we maintain order of content.

Reply:
I can't think of how this affects maintaining order of content, as the important thing is that `VectorisedJobPointer` points to the correct locations. See how pointers are added [here](https://github.com/marqo-ai/marqo/pull/373/files#diff-bc6ddb28e3809766b86804afac01ac632de1e691377c5d3f43d0333b97ba79faR1497). 

What are the situations where you think this could fail? I can make some extra tests if you have ideas